.github/workflows/tx-hammer-plugins.yml

```diff
@@ -0,0 +1,93 @@
+name: Translations
+
+on:
+  workflow_call:
+    inputs:
+      plugin_name:
+        description: 'Name of the Hammer plugin. (Not the repository name)'
+        required: true
+        type: 'string'
+    secrets:
+      tx_token:
+        description: 'API token for the Transifex.'
+        required: true
+      gh_token:
+        description: 'GitHub token with permissions to create a pull request'
+        required: true
+env:
+  branch_name: "tx-${{ github.run_id }}"
+jobs:
+  tx-update:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.plugin_name  }}
+
+    steps:
+      - name: Get hammer-cli
+        uses: actions/checkout@v4
+        with:
+          repository: theforeman/hammer-cli
+          ref: master
+          path: hammer-cli
+
+      - name: Get ${{ github.repository }}
+        uses: actions/checkout@v4
+        with:
+          path: ${{ inputs.plugin_name  }}
+      - name: Install Transifex dependencies
+        run: |
+          sudo apt-get -y install gettext
+
+      - name: "Install Transifex"
+        run: |
+          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
+          sudo mv tx /usr/local/bin/
+          sudo chown -R runner:runner /usr/local/bin
+
+      - name: Configure plugin
+        run: |
+          echo "gemspec path: '../hammer-cli', name: 'hammer_cli'" > Gemfile.local.rb
+          mkdir -p ./.tx
+          echo "[main]" > ./.tx/config
+          echo "host = https://rest.api.transifex.com" >> ./.tx/config
+          echo "token = ${{ secrets.tx_token  }}" >> ./.tx/config
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "2.7"
+          bundler-cache: true
+          working-directory: ${{ inputs.plugin_name  }}
+      - name: Set up Git
+        run: |
+          git config user.name "Foreman Translation action"
+          git config user.email "noreply@foreman.example.com"
+
+          git checkout -b $branch_name
+
+          echo ".tx/*" >> .gitignore
+          echo "vendor/*" >> .gitignore
+
+      - name: Pull translations
+        env:
+          TX_TOKEN: ${{ secrets.tx_token  }}
+        run: |
+          bundle exec rake tx:update
+
+      - name: Check for the changes
+        run: |
+          if ! [ -n "$(git diff ${{ github.ref_name }})" ]; then
+            echo '###'
+            echo '### No changes.'
+            echo '### Exiting ...'
+            echo '###'
+            exit 1
+          fi
+
+      - name: Push changes and create a pull request
```

Comment:
In packaging we have this bit: https://github.com/theforeman/foreman-packaging/blob/55448ce7ffbee75b9edd16f54b22030580fada5d/.github/workflows/bump_packages.yml#L64-L71

We know that one works well and also deals with an already existing branch + PR.

Reply:
Yeah, I can use that, I wasn't sure what's the policy for using actions from the "outside".